tiledb/sm/rest/curl.h

```diff
@@ -59,6 +59,21 @@ namespace sm {
  * presented by this class is not threadsafe either. See
  * https://curl.haxx.se/libcurl/c/threadsafe.html
  */
+
+/**
+ * Wraps opaque user data to be invoked with a header callback.
+ */
+struct HeaderCbData {
+  /*The output of parse::rest_components from url -> array_ns:array_uri */
+  std::string uri;
+
+  /* A pointer to the map in REST client caching the redirections*/
+  std::unordered_map<std::string, std::string>* redirect_uri_map;
+
+  /*A pointer to the lock attached to the shared resource of the cache map*/
```

Comment:
- Our in-code docs use the syntax `/** */` instead of `/* */`, please update these to match.
- Ensure there is a space after `/**` and before `*/` in block comments like these.